src/main/java/com/playmonumenta/networkrelay/RemotePlayerManager.java

```diff
@@ -33,10 +36,17 @@ public static class RemotePlayer {
 		public RemotePlayer(Player player, boolean isOnline) {
 			mUuid = player.getUniqueId();
 			mName = player.getName();
-			mShard = getShardName() != null ? getShardName() : NetworkRelayAPI.getShardName();
-			mIsHidden = !isLocalPlayerVisible(player);
+			mIsHidden = !internalPlayerVisibleTest(player);
 			mIsOnline = isOnline;
+			mShard = getShardName();
 			mPluginData = new ConcurrentHashMap<>();
+
+			// Gather Plugin data
```

Comment:
Moved plugin data collection here. It should only update when updated by the refresh method.